pkg/controllers/repositoryserver/utils.go

```diff
@@ -0,0 +1,268 @@
+// The container spec will include the matching volumeMount.
+func volumeMountSpecForName(podSpec corev1.PodSpec, podOverride map[string]interface{}, certName string) (string, bool) {
+	if certName == "" {
+		return "", false
+	}
+	for _, ctr := range podSpec.Containers {
+		for _, mount := range ctr.VolumeMounts {
+			if mount.SubPath != certName {
+				continue
+			}
+			mountList := []corev1.VolumeMount{mount}
+			ctr := &corev1.Container{
+				Name:         "container",
+				VolumeMounts: mountList,
+			}
+			podOverride["containers"] = []corev1.Container{*ctr}
+			return mount.Name, true
+		}
+	}
+	return "", false
+}
+
+// volumeSpecForName adds a pod's volume spec to the override map
+// if the volume's name matches with a given volumeName
+func volumeSpecForName(podSpec corev1.PodSpec, podOverride map[string]interface{}, volumeName string) {
+	for _, vol := range podSpec.Volumes {
+		if vol.Name == volumeName {
+			podOverride["volumes"] = []corev1.Volume{vol}
+			return
+		}
+	}
+}
+
+func addTLSCertConfigurationInPodOverride(podOverride *map[string]interface{}, tlsCertSecretName string) error {
+	podSpecBytes, err := json.Marshal(*podOverride)
+	if err != nil {
+		return errors.Wrap(err, "Failed to marshal Pod Override")
+	}
+
+	var podOverrideSpec corev1.PodSpec
+	if err := json.Unmarshal(podSpecBytes, &podOverrideSpec); err != nil {
+		return errors.Wrap(err, "Failed to unmarshal Pod Override Spec")
+	}
+
+	podOverrideSpec.Volumes = append(podOverrideSpec.Volumes, corev1.Volume{
+		Name: tlsCertVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: tlsCertSecretName,
+			},
+		},
+	})
+
+	if len(podOverrideSpec.Containers) == 0 {
+		podOverrideSpec.Containers = append(podOverrideSpec.Containers, corev1.Container{
+			Name: "container",
+		})
+	}
+
+	podOverrideSpec.Containers[0].VolumeMounts = append(podOverrideSpec.Containers[0].VolumeMounts, corev1.VolumeMount{
+		Name:      tlsCertVolumeName,
+		MountPath: tlsCertDefaultMountPath,
+	})
+
+	podSpecBytes, err = json.Marshal(podOverrideSpec)
+	if err != nil {
+		return errors.Wrap(err, "Failed to marshal Pod Override Spec")
+	}
+
+	if err := json.Unmarshal(podSpecBytes, podOverride); err != nil {
+		return errors.Wrap(err, "Failed to unmarshal Pod Override")
+	}
+
+	return nil
+}
+
+func getPodOptions(namespace string, podOverride map[string]interface{}, svc *corev1.Service) *kube.PodOptions {
+	uidguid := int64(0)
+	nonRootBool := false
+	return &kube.PodOptions{
+		Namespace:     namespace,
+		GenerateName:  fmt.Sprintf("%s-", repoServerPod),
+		Image:         consts.KanisterToolsImage,
+		ContainerName: repoServerPodContainerName,
+		Command:       []string{"bash", "-c", "tail -f /dev/null"},
+		PodOverride:   podOverride,
+		Labels:        map[string]string{repoServerServiceNameKey: svc.Name},
+		PodSecurityContext: &corev1.PodSecurityContext{
+			RunAsUser:    &uidguid,
+			RunAsNonRoot: &nonRootBool,
+		},
```

Comment:
Let's add a TODO here to make sure we are in sync with latest changes in tools pod.

Reply:
I did not understand this comment, I already have made the changes related to podSecurityContext as we discussed earlier. Do I need to do anything else?